reana_commons/utils.py

```diff
@@ -56,3 +60,40 @@ def get_user_analyses_dir(org, user):
     :return: Path to the user's analyses directory.
     """
     return fs.path.join(org, user, 'analyses')
+
+
+def calculate_hash_of_dir(directory, verbose=0):
+    """Calculate hash of directory."""
+    SHAhash = hashlib.md5()
+    if not os.path.exists(directory):
+        return -1
+
+    try:
+        for root, dirs, files in os.walk(directory):
+            for names in files:
+                filepath = os.path.join(root, names)
+                try:
+                    f1 = open(filepath, 'rb')
+                except Exception:
+                    # You can't open the file for some reason
+                    f1.close()
```

Comment:
Should we invalidate the cache if this happens? The file you can not open might be changed in the next execution or it can even not be present

Reply:
If it's changed, it won't be cache hit. If it's not opened in 2 consecutive calculations then it'll have the same md5 ftom the other possible files and in this case yes, we should not store it in the cache.